src/Nox.Types/Types/Length/Length.cs

```diff
@@ -1,9 +1,126 @@
 namespace Nox.Types;
 
+/// <summary>
+/// Represents a Nox <see cref="Length"/> type and value object.
+/// </summary>
+public sealed class Length : ValueObject<QuantityValue, Length>
+{
+    private const int QuantityValueDecimalPrecision = 6;
+
+    public LengthTypeUnit Unit { get; private set; } = LengthTypeUnit.Meter;
+
+    public Length() { Value = 0; }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="Length"/> object in meters.
+    /// </summary>
+    /// <param name="value">The value to create the <see cref="Length"/> with</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static Length FromMeters(QuantityValue value)
+        => From(value, LengthTypeUnit.Meter);
+
+    /// <summary>
+    /// Creates a new instance of <see cref="Length"/> object in feet.
+    /// </summary>
+    /// <param name="value">The origin value to create the <see cref="Length"/> with</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static Length FromFeet(QuantityValue value)
+        => From(value, LengthTypeUnit.Foot);
+
     /// <summary>
-    /// Represents a Nox <see cref="Length"/> type and value object.
+    /// Creates a new instance of <see cref="Length"/> object in meters.
     /// </summary>
-    /// <remarks>Placeholder, needs to be implemented</remarks>
-    public sealed class Length : ValueObject<QuantityValue, Length>
+    /// <param name="value">The value to create the <see cref="Length"/> with</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public new static Length From(QuantityValue value)
+        => From(value, LengthTypeUnit.Meter);
+
+    /// <summary>
+    /// Creates a new instance of <see cref="Length"/> object with the specified <see cref="LengthTypeUnit"/>.
+    /// </summary>
+    /// <param name="value">The value to create the <see cref="Length"/> with</param>
+    /// <param name="unit">The <see cref="LengthTypeUnit"/> to create the <see cref="Length"/> with</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static Length From(QuantityValue value, LengthTypeUnit unit)
     {
+        var newObject = new Length
+        {
+            Value = Round(value),
+            Unit = unit,
+        };
+
+        var validationResult = newObject.Validate();
+
+        if (!validationResult.IsValid)
+        {
+            throw new TypeValidationException(validationResult.Errors);
+        }
+
+        return newObject;
     }
+
+    /// <summary>
+    /// Validates a <see cref="Length"/> object.
+    /// </summary>
+    /// <returns>true if the <see cref="Length"/> value is valid.</returns>
+    internal override ValidationResult Validate()
+    {
+        var result = Value.Validate();
+
+        if (Value < 0 && !double.IsNaN((double)Value) && !double.IsInfinity((double)Value))
```

Comment:
I've added the NaN and Infinity check because in case when Value was NaN or Infinity, we would receive two errors:

`    Nox.Types.ValidationFailure
    {
        ErrorMessage = "Could not create a Nox type as value NaN is not allowed.", 
        Variable = "Value"
    }, 
    Nox.Types.ValidationFailure
    {
        ErrorMessage = "Could not create a Nox Length type as negative length value NaN is not allowed.", 
        Variable = "Value"
    }`

If this is expected, I can remove the additional checks.